DSharpPlus/Entities/Channel/DiscordChannel.cs

```diff
@@ -1101,6 +1101,54 @@ public Permissions PermissionsFor(DiscordMember mbr)
         return perms;
     }
 
+    /// <summary>
+    /// Calculates permissions for a given role.
+    /// </summary>
+    /// <param name="role">Role to calculate permissions for.</param>
+    /// <returns>Calculated permissions for a given role.</returns>
+    public Permissions PermissionsFor(DiscordRole role)
+    {
+        if (this.IsThread)
+        {
+            return this.Parent.PermissionsFor(role);
+        }
+        
+        if (this.IsPrivate || this.Guild is null)
+        {
+            return Permissions.None;
+        }
```

Comment:
wouldn't we also want to check whether the role and channel belong to the same guild